state/allcollections.go

```diff
@@ -246,8 +246,12 @@ func allCollections() collectionSchema {
 		// -----
 
 		// These collections hold information associated with applications.
-		charmsC:       {},
-		applicationsC: {},
+		charmsC: {},
```

Comment:
I stopped here for tonight. I may try to give it another look tomorrow.
Would it be reasonable to document *why* we want a given index somewhere?